components/alert/alert-toast.js

```diff
@@ -0,0 +1,183 @@
+import '../button/button-icon.js';
+import '../button/button-subtle.js';
+import '../colors/colors.js';
+import './alert.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+
+const reduceMotion = matchMedia('(prefers-reduced-motion: reduce)').matches;
+
+const states = {
+	CLOSED: 'closed', // the toast is closed
+	CLOSING: 'closing', // the close animation is running
+	PREOPENING: 'preopening', // a pause before running the opening animation because transitions won't run when changing from 'diplay: none' to 'display: block'
+	OPENING: 'opening', // the opening animation is running
+	OPEN: 'open' // the toast is open
+};
+
+class AlertToast extends LitElement {
+
+	static get properties() {
+		return {
+			buttonText: { type: String, attribute: 'button-text' },
+			hideCloseButton: { type: Boolean, attribute: 'hide-close-button' },
+			noAutoClose: { type: Boolean, attribute: 'no-auto-close' },
+			open: { type: Boolean, reflect: true },
+			subtext: { type: String },
+			type: { type: String, reflect: true },
+			_state: { type: String, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}
+
+			.d2l-alert-toast-container {
+				border-radius: 0.3rem;
+				bottom: 1.5rem;
+				box-shadow: 0 0.1rem 0.6rem 0 rgba(0,0,0,0.10);
+				display: none;
+				left: 0;
+				margin: 0 auto;
+				max-width: 600px;
+				position: fixed;
+				right: 0;
+				width: 100%;
+				z-index: 10000;
+			}
+
+			:host(:not([_state="closed"])) .d2l-alert-toast-container {
+				display: block;
+			}
+
+			:host([_state="opening"]) .d2l-alert-toast-container,
+			:host([_state="closing"]) .d2l-alert-toast-container {
+				transition-duration: 250ms;
+				transition-property: transform, opacity, visibility;
+				transition-timing-function: ease-in;
+			}
+
+			:host([_state="preopening"]) .d2l-alert-toast-container,
+			:host([_state="closing"]) .d2l-alert-toast-container {
+				opacity: 0;
+				transform: translateY(0.5rem);
+				visibility: hidden;
+			}
+
+			:host([_state="opening"]) .d2l-alert-toast-container {
+				opacity: 1;
+				transform: translateY(0);
+				visibility: visible;
+			}
+
+			.d2l-alert-toast {
+				animation: none;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.hideCloseButton = false;
+		this.noAutoClose = false;
+		this.open = false;
+		this._state = states.CLOSED;
+	}
+
+	get _state() {
+		return this.__state;
+	}
+
+	set _state(val) {
+		const oldVal = this.__state;
+		if (oldVal !== val) {
+			this.__state = val;
+			this.requestUpdate('_state', oldVal);
+			this._stateChanged(val, oldVal);
+		}
+	}
+
+	get open() {
+		return this._open;
+	}
+
+	set open(val) {
+		const oldVal = this._open;
+		if (oldVal !== val) {
+			this._open = val;
+			this.requestUpdate('open', oldVal);
```

Comment:
Is this `requestUpdate()` call needed?  It looks like `render` does not depend on `open` value.

Reply:
^ In experimenting, I found this is necessary in order for Lit to reflect the property update to the `open` attribute.  Otherwise, it would not be needed.